archive_communication/api-requirements.md

```diff
@@ -0,0 +1,324 @@
+    <td class="tg-lboi">
+        <ul>
+            <li>JSONSchema preferred</li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">6</td>
+    <td class="tg-lboi">Related objects are easily findable from a given object</td>
+    <td class="tg-lboi">Objects contain resource IDs or resource URIs for related objects.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li> Supports use case <a href="api-use-cases.md"> Data inventory – new project</a>
+            <li><a href="#related-objects">See example</a></li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">7</td>
+    <td class="tg-lboi">Consistent response status codes</td>
+    <td class="tg-lboi">API responses should contain appropriate, standardized status codes.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>
+                <a href="https://datatracker.ietf.org/doc/html/rfc2616"> 
+                    Proposed standard: IETF RFC 2616
+                </a>
+            </li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">8</td>
+    <td class="tg-lboi">API endpoint documentation</td>
+    <td class="tg-lboi">Public endpoints should be documented with expected request params and sample responses.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi"></td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">9</td>
+    <td class="tg-lboi">Filtering results</td>
+    <td class="tg-lboi">Results for API calls can be filtered via query string or filters specified in the request body.</td>
+    <td class="tg-lboi">Nice to have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li><a href="#filtering-results">See example</a></li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">10</td>
+    <td class="tg-lboi">Searching by BCDC identifier</td>
+    <td class="tg-lboi">A user should be able to find a resource using the resource's BCDC identifier</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>
+                Supports use case: <a href="api-use-cases.md"> Data inventory – registered projects </a>
+            </li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">11</td>
+    <td class="tg-lboi">Searching by metadata</td>
+    <td class="tg-lboi">Search for data sets using parameters like grant number, species, technique, etc.</td>
+    <td class="tg-lboi">Nice to have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li><a href="#searching"> See example </a></li>
+            <li>Supports use case: <a href="api-use-cases.md"> Search – filter for projects</a></li>
+            <li> Supports use case: <a href="api-use-cases.md">Search – filter for samples</a></li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">12</td>
+    <td class="tg-lboi">Validation exceptions</td>
```

Comment:
@pbishwakarma a dataset that doesn't pass validation wouldn't be public, hence it wouldn't be part of the response.

@ltuite96 @mariahkenney there is a potential to query by dataset ID or UUID and retrieve the status of the ingestion.